packages/frontend/app/routes/tenants.$tenantId.tsx

```diff
@@ -0,0 +1,322 @@
+  await checkAuthAndRedirect(request.url, cookies)
+
+  const tenantId = params.tenantId
+  const result = z.string().uuid().safeParse(tenantId)
+  if (!result.success) {
+    throw json(null, { status: 400, statusText: 'Invalid tenant ID.' })
+  }
+
+  const tenant = await getTenantInfo(request, { id: result.data })
+  if (!tenant)
+    throw json(null, { status: 404, statusText: 'Tenant not found.' })
+
+  const me = await whoAmI(request)
+  return json({ tenant, me })
+}
+
+export default function ViewTenantPage() {
+  const { tenant, me } = useLoaderData<typeof loader>()
+  const response = useActionData<typeof action>()
+  const navigation = useNavigation()
+  const [formData, setFormData] = useState<FormData>()
+
+  const submit = useSubmit()
+  const dialogRef = useRef<ConfirmationDialogRef>(null)
+
+  const isSubmitting = navigation.state === 'submitting'
+
+  const submitHandler = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    setFormData(new FormData(event.currentTarget))
+    dialogRef.current?.display()
+  }
+
+  const onConfirm = () => {
+    if (formData) {
+      submit(formData, { method: 'post' })
+    }
+  }
+
+  return (
+    <div className='pt-4 flex flex-col space-y-4'>
+      <div className='flex flex-col rounded-md bg-offwhite px-6'>
+        <PageHeader className='!justify-end'>
+          <Button aria-label='go back to tenants page' to='/tenants'>
+            Go to tenants page
+          </Button>
+        </PageHeader>
+        <div className='grid grid-cols-1 py-3 gap-6 md:grid-cols-3 border-b border-pearl'>
+          <div className='col-span-1 pt-3'>
+            <h3 className='text-lg font-medium'>General Information</h3>
+            <p className='text-sm'>
+              Created at {new Date(tenant.createdAt).toLocaleString()}
+            </p>
+            <ErrorPanel errors={response?.errors.general.message} />
+          </div>
+          <div className='md:col-span-2 bg-white rounded-md shadow-md'>
+            <Form method='post' replace preventScrollReset>
+              <fieldset disabled={isSubmitting}>
+                <div className='w-full p-4 space-y-3'>
+                  <Input type='hidden' name='id' value={tenant.id} />
+                  <Input
+                    label='Tenant ID'
+                    name='tenantId'
+                    value={tenant.id}
+                    disabled
+                    readOnly
+                  />
+                  <Input
+                    label='Public Name'
+                    name='publicName'
+                    defaultValue={tenant.publicName ?? undefined}
+                    error={response?.errors.general.fieldErrors.publicName}
+                  />
+                  <Input
+                    label='Email'
+                    name='email'
+                    defaultValue={tenant.email ?? undefined}
+                    error={response?.errors.general.fieldErrors.email}
+                  />
```

Comment:
Is it possible to have the email field error be inline?

![Screenshot 2025-02-12 at 15 19 08](https://github.com/user-attachments/assets/b85391c6-e0f3-4ee4-8e05-3b724a518324)


Something like:


![Screenshot 2025-02-12 at 15 20 17](https://github.com/user-attachments/assets/be4fb5ee-d189-4a5d-a73b-9c2820236609)